resize.js

```diff
@@ -63,296 +35,161 @@ function logRequest(req, res, time) {
+
+    image.checkCacheOrCreate(params, res);
+  },
+  checkCacheOrCreate(params, res) {
+    // Check if it exists in the cache
+    selectImage.get([params.fileName, params.resolutionX, params.resolutionY, params.fit, parsing.supportedFileType(params.fileType)], function (err, data) {
+      if (!err && data) {
+        // It is in the cache, so redirect to there
+        return helpers.send307DueToCache(res, params, data.url);
       }
-      var originalRatio = size.width / size.height;
-      var newRatio = resolutionX / resolutionY;
-
-      var resizeFactor;
-      var cropX = 0;
-      var cropY = 0;
-      var cropWidth = size.width;
-      var cropHeight = size.height;
 
-      if (fit === 'crop') {
-        if (originalRatio > newRatio) {
-          resizeFactor = size.height / resolutionY;
-          cropWidth = size.width / resizeFactor;
-          cropHeight = resolutionY;
-          cropX = (cropWidth - resolutionX) / 2;
-        }
-        else {
-          resizeFactor = size.width / resolutionX;
-          cropWidth = resolutionX;
-          cropHeight = size.height / resizeFactor;
-          cropY = (cropHeight - resolutionY) / 2;
-        }
+      // It does not exist in the cache, so generate and upload
+      image.encodeAndUpload(params, res);
+    });
+  },
+  encodeAndUpload (params, res) {
+    var file = config.get('originals_dir') + '/' + params.fileName;
+    fs.exists(file, function (exists) {
+      if (!exists) {
+        log.log('warn', `File ${params.fileName} was requested but did not exist`);
+        return helpers.send404(res, params.fileName);
       }
 
-      var workImageClient = gm(file)
-        .options({imageMagick: true})
-        .autoOrient();
-
-      if (resizeFactor) {
-        workImageClient = workImageClient.resize(cropWidth, cropHeight).crop(resolutionX, resolutionY, cropX, cropY);
-      } else {
-        workImageClient = workImageClient.resize(resolutionX, resolutionY);
-      }
-      workImageClient.stream(fileType, function (err, stdout, stderr) {
-        var r = stdout.pipe(res);
-        r.on('finish', function () {
-          // This is to close the result while a background job will continue to process
-          log('info', 'Finished sending a converted image');
-          res.end();
+      // Get the image and resize it
+      res.writeHead(200, {'Content-Type': parsing.supportedFileType(params.fileType)});
+
+      // These files have already been oriented!
+      correctlyResize(file, params, (resized) => {
+        resized.stream(params.fileType, function (err, stdout) {
+          var r = stdout.pipe(res);
+          r.on('finish', function () {
+            // This is to close the result while a background job will continue to process
+            log.log('info', 'Finished sending a converted image');
+            res.end();
+          });
```

Comment:
This callback is a no-op: when piping, the destination stream closes when the source stream closes. Additionally, `pipe` returns the destination stream so this is effectively `res.on('finish', () => res.end())`